rtlib/src/DisegniSegmenter.cpp

```diff
@@ -0,0 +1,167 @@
+#include "rt/DisegniSegmenter.hpp"
+
+static const int INT_MINI = std::numeric_limits<int>::min();
+static const int INT_MAXI = std::numeric_limits<int>::max();
+using namespace cv;
+
+// Takes an Input Image and sets it into your private image member
+void DisegniSegmenter::setInputImage(Mat i) { inputImage_ = i; }
+
+// Conducts all the private class functions to keep the
+// ImageSegmentationMain.cpp concise
+std::vector<Mat> DisegniSegmenter::compute()
+{
+    auto labeledImage = watershed_image_();
+    return split_merge_(labeledImage);
+}
+
+/*
+ * In the study of Image Processing, a watershed is a transformation defined on
+a grayscale image. The tool treats the image like a topographic map, with the
+brightness of each point representing its height. The tool then finds the lines
+that run along the ridges/boundaries of the points of interest. Finally, it
+outputs an image with only the points of interest visible.
+ */
```

Comment:
Since this code largely comes from the OpenCV tutorial, put a link to that here.